normalize-own-data/quantile_normalize_own_data.Rmd

```diff
@@ -0,0 +1,207 @@
+from the refine.bio website, you may wish to quantile normalize it using the
+reference distribution we've used in refine.bio.
+
+You can read more about the reference distribution and how it is generated in 
+the [quantile normalization section](http://docs.refine.bio/en/latest/main_text.html#quantile-normalization) 
+of our [documentation](http://docs.refine.bio/en/latest/).
+
+In this example, we'll demonstrate how to obtain the reference distribution
+from the refine.bio API and apply it to your own data in R.
+
+### Dataset
+
+We used [`GSE39842`](https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE39842) 
+in another module of the GitHub repository of examples: [`AlexsLemonade/refinebio-examples/compare-processing`](https://github.com/AlexsLemonade/refinebio-examples/tree/master/compare-processing).
+
+The submitters used RMA to process their data. 
+We have already converted the gene identifiers to Ensembl gene ids and can
+obtain the TSV file with these data from GitHub, placing it in the `data` 
+directory we create below with `bash`.
+
+```{bash eval=FALSE}
+mkdir data
+wget --directory-prefix=data https://github.com/AlexsLemonade/refinebio-examples/raw/c97cad6a8342d9bb54f120dbd7c7f12b03e3f799/compare-processing/data/GSE39842_RMA_converted.tsv
+```
+
+## Obtaining a refine.bio QN target 
+
+We'll use the terms "QN target" and "reference distribution" interchangeably to
+refer to the vector of values we use to quantile normalize `GSE39842`.
+
+We can query the refine.bio API like so:
+
+```
+https://api.refine.bio/qn_targets/?organism=<SCIENTIFIC_NAME>
+```
+
+where we would replace `<SCIENTIFIC_NAME>` with the name of the species we are
+interested in.
+
+We'd query the API for the zebrafish QN target like so:
+
+```
+https://api.refine.bio/qn_targets/?organism=DANIO_RERIO
+```
+
+Here, we'll use the `jsonlite` R package to extract the information we need to
+obtain the TSV file with the zebrafish reference distribution values. 
+
+```{r}
+qn_target_info <- jsonlite::fromJSON("https://api.refine.bio/qn_targets/?organism=DANIO_RERIO")
+```
+
+We need the `s3_url`, which tells us the link where the file is available for
+download from.
+We can read this data into R with the URL directly later on in this notebook.
+
+```{r}
+qn_url <- qn_target_info$s3_url
+```
+
+This is all the information we need to continue.
+
+```{r}
+rm(qn_target_info)
+```
+
+## Set up 
+
+We need the [`preprocessCore`](https://bioconductor.org/packages/release/bioc/html/preprocessCore.html) 
+package to perform quantile normalization. 
+
+```{r}
+if (!("preprocessCore" %in% installed.packages())) {
+  BiocManager::install("preprocessCore", update = FALSE)
+}
+```
+
+```{r}
+`%>%` <- dplyr::`%>%`
+```
+
+### Read in data
+
+First, the gene expression matrix.
+
+```{r}
+expression_file <- file.path("data", "GSE39842_RMA_converted.tsv")
+expression_df <- readr::read_tsv(expression_file)
+```
```

Comment:
Maybe put a space here? Also I'm assuming based on these steps that `preprocessCore` wants the data as a matrix and that's why you are doing these steps? Could put a quick sentence to give the readers a heads up on `preprocessCore`'s format requirements.